src/hooks/wallets/trezor.hook.ts

```
@@ -63,15 +76,44 @@ export function useTrezor(): TrezorInterface {
     handlePayloadError('Trezor not connected', result.payload.error);
   }
 
-  async function signMessage(msg: string, wallet: TrezorWallet, addressIndex: number): Promise<string> {
+  async function fetchAddresses(
+    wallet: TrezorWallet,
+    startIndex: number,
+    count: number,
+    bitcoinAddressType: BitcoinAddressType,
+  ): Promise<string[]> {
+    try {
+      const addresses = [];
+      for (let i = startIndex; i < startIndex + count; i++) {
+        const result =
+          wallet === WalletType.TREZOR_BTC
+            ? await TrezorConnect.getAddress({
+                path: KeyPath.BTC(bitcoinAddressType).address(i),
+                showOnTrezor: false,
+              })
+            : await TrezorConnect.ethereumGetAddress({ path: KeyPath.ETH.address(i), showOnTrezor: false });
+        if (result.success) addresses.push(result.payload.address);
+      }

```

Maybe we should throw an error, if it wasn't a success?